src/vmm/src/devices/virtio/queue.rs

```diff
@@ -106,8 +106,8 @@ impl<'a, M: GuestMemory> DescriptorChain<'a, M> {
             return None;
         }
 
-        let desc_head = mem.checked_offset(desc_table, (index as usize) * 16)?;
-        mem.checked_offset(desc_head, 16)?;
+        // There's no need for checking as we already validated the descriptor table and index bounds.
+        let desc_head = desc_table.unchecked_add((index as usize) * 16);
```

Comment:
think it needs to be
```suggestion
        let desc_head = desc_table.unchecked_add((index as u64) * 16);
```
for the compiler to be happy here

Reply:
Sorry it was my mistake. I've fixed it and added doc for `DescriptorChain::checked_new`.